src/core/Target.mli

```diff
@@ -0,0 +1,146 @@
+}
+[@@deriving show]
+(** A "normal" semgrep target, comprising source code (or, for
+   regex/generic, arbitrary text data) to be executed. See also {!Xtarget.t},
+   an augmented version which also has the contents. *)
+
+and lockfile = {
+  path : path;
+  kind : Lockfile_kind.t;
+      (** The type of lockfile this is. Analogous to analyzer for a source code
+          target. *)
+  manifest : manifest option;
+      (** Optionally, a manifest file associated with this lockfile. *)
+}
+[@@deriving show]
+(** A lockfile to be used during matching. See also {!Lockfile_xtarget.t}, an
+    augmented version with the contents of the lockfile. *)
+
+and manifest = {
+  path : path;
+  kind : Manifest_kind.t;
+      (** The type of manifest this is. Analogous to analyzer for a source code
+        target. *)
+}
+[@@deriving show]
+(** A manifest file to be used during matching. See also
+    {!Lockfile_xtarget.manifest}, which also has the contents. *)
+
+and path = {
+  origin : Origin.t;
+      (** The origin of the data as is relevant to the user. This could be,
+          e.g., a relative (from the project root) path to a file, a git object
+          and associated information, or anything else a Origin.t can
+          designate.
+
+          This should be used when reporting a location to the user. *)
+  internal_path_to_content : Fpath.t;
+      (** The path to a file which contains the data to be scanned. This could
+          be the same as the origin, if the origin is a path to a regular file
+          (or an absolute path to the same), or it could be a tempfile. This
+          should be used to obtain the contents of the target, but not for
+          reporting to the user, other than possibly for debugging purposes. *)
+}
+[@@deriving show]
+(** Information about where a target from for both the purpose of
+   {ul
+    {- informing the user: [origin]}
+    {- obtaining the contents: [internal_path_to_content]}
+  } *)
+
+(* deriving eq appears to not work on mutual definitions with and. See also
+   <https://github.com/ocaml-ppx/ppx_deriving/issues/272>, for deriving make,
+   which may be relevant. Therefore, we just define it here. *)
+val equal_path : path -> path -> bool
+
+val mk_code :
+  ?lockfile:lockfile ->
+  Xlang.t ->
+  Semgrep_output_v1_t.product list ->
+  Origin.t ->
+  code
+(** [mk_code analyzer products origin] is the target
+      location for a source code target originating from [origin] to be
+      analyzed with [analyzer] for [products]. If [lockfile] is specified then
+      it shall be used as the associated lockfile if dependency patterns are
+      to be ran.
+
+      This function should be generally preferred over creating a record
+      directly, since it can peform actions which may be required when creating
+      a target from certain types of origins, such as generating a tempfile.
+ *)
+
+val mk_lockfile : ?manifest:manifest -> Lockfile_kind.t -> Origin.t -> lockfile
```

Comment:
and mk_lockfile_target (or sca_target).

Reply:
I think this should still be `mk_lockfile` for a couple reasons:
- I think this should parallel the name of the constructor for Target.t. Currently that is `Lockfile`, carried over from Matthew's recent PR.
- Given these are always used like Target.mk_regular or Target.mk_lockfile I don't see the point of including target again in the function name (similar to Target.path being preferred to Target.target_path).